src/extension.ts

```diff
@@ -414,12 +415,15 @@ class ExtensionManager implements vscode.Disposable {
     await telemetry.deactivate();
   }
 
-  async configureInternal(trigger: ConfigureTrigger, cmt: CMakeTools): Promise<void> {
+  async configureExtensionInternal(trigger: ConfigureTrigger, cmt: CMakeTools): Promise<void> {
     if (!await this._ensureActiveConfigurePresetOrKit(cmt)) {
       return;
     }
-
-    await cmt.configureInternal(trigger, [], ConfigureType.Normal);
+    if (trigger === ConfigureTrigger.configureOnOpen && !cmt.workspaceContext.config.configureOnOpen) {
```

Comment:
I think this check could be moved into `configureInternal` and you could remove `ConfigureType.Auto` from the enum.

Reply:
Removing the enum from CMakeDriver was more straightforward, as we have only one driver of type `CMakeFileApiDriver` and we could add a flag `_isConfiguredatLeastOnce` to it, and this flag shows if we have called configure at least once or not.
but in the `configureInternal` in CMakeTools, we have to check if the driver instance is type of `CMakeFileApiDriver` and then check if it's the first time or not. I found that to be a more confusing check, compare to having `ConfigureType.Auto`. 
I changed the code to remove `ConfigureType.Auto`. Let me know which one you think is more clear. I added some comments for clarity, in case we go for the second approach.

Reply:
I made the change. the check is a little long but it works.

Reply:
Do you actually need to check the `type`?  I would think the `trigger` check makes that unnecessary.

You could also move some of the extra checks into an abstract method that returns true|false if it would be acceptable to load a cached configuration.  (e.g. drv is CMakeFileApiDriver && !configureOnOpen && !isConfiguredAtLeastOnce)

Reply:
I added a simpler check, i.e. `drv.shouldUseCachedConfiguration(trigger)` that calls to the function `isCacheConfigSupported` and this approach won't need to check the instance of the driver directly.